sdks/Cordova.md

```diff
@@ -0,0 +1,136 @@
+# Apache Cordova AeroGear Services
+
+## Motivation
+
+Provide set of the SDK's that can be used for Cordova based development.
+Provide best performance and user experience on the platform.
+
+## Proposed solution 1
+
+Pure javascript solution for the mobile configuration and orchestration 
+that will allow cooperability between JavaScript and Native.
+Configuration parser and other pure JS code willnot require any native implementation and can be reused for cordova/react native implementations. Individual SDK will may by default be implemented in pure JavaScript or as Cordova/React plugins. 
+
+## Advantages of the solution 1
+
+- Significantly reduced effort to develop and maintain Cordova Support by reusing available components.
+- Aligned with the way cordova works (single plugin to do specific *native* job, integrated with others in JavaScript)
+- Ability to reuse existing cordova plugins that perform native tasks (push notifications, sync, auth)
+- Ability to wrap existing JavaScript solutions and use them without 
+- Ability to select and use only elements that are needed in the app.
+
+See investigation details bellow for more details
+
+## Proposed solution 2
+
+Pure JavaScript implementations (JavaScript SDK) that will we written for the web and abstract from browser support.
+This means that specific browser functionalities that may not be available for React or Cordova will be provided to enable support for this platforms. JS SDK will be working with Web, Cordova and ReactNative - each platform will provide wrappers for specific apis if needed. This aproach will not allow us to reuse native implementations, but it will make Cordova, Web and ReactNative SDK's more consistent. Implementation could detect platform and load specific ReactNative/Cordova/Browser wrappers.
```

Comment:
I like this idea if I understand 👍 

So you mean something like this?

```js
const platform = require('src/platform')

if (platform.isReactNative()) {
  // Do RN implementation
} else if (platform.isCordova()) {
  // Do Cordova implementation
} else {
  // Web/Electron etc...
}
```

It would make the bundle a little larger to load, but also has all those advantages, and easier documentation due to a unified API.

Reply:
@evanshortiss - You are right. I'm investigating now using webpack that will allow us to have single codebase that will produce different flavors with different code. Your code snipped will still work fine but we could have that if on build time and produce separate artifacts for the different platforms.

Reply:
That'd be much better since it'd have smaller JS bundle size 👍